### PR TITLE
Bug 1830914 - Fixes to make running taskgraph locally easier

### PR DESCRIPTION
**Pull request by ahal**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1830914

### CODE REVIEW

File: taskcluster/test/params/main-repo-ship-release.yml [hunk 1]
```diff
@@ -167,4 +167,4 @@ repository_type: git
 shipping_phase: ship
 target_tasks_method: ship
 tasks_for: action
-version: 107.0.4
+version: 113.0a1
```
Comment (JohanLorenzo) [inline taskcluster/test/params/main-repo-ship-release.yml:170]: Nit: Let's use a release-looking version number here.

File: taskcluster/android_taskgraph/parameters.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    }
+
+
+extend_parameters_schema(
+    {
+        Required("pull_request_number"): Any(All(int, Range(min=1)), None),
+        Required("release_type"): str,
+    },
+    defaults_fn=get_defaults,
+)
```
Comment (JohanLorenzo) [inline taskcluster/android_taskgraph/parameters.py:22]: Nit: Let's remove the call from there:

https://github.com/mozilla-mobile/firefox-android/blob/191d5fd17d0a9d610770db7668ea6563d715effa/taskcluster/android_taskgraph/__init__.py#L19-L23

We may want to add `next_version` too here.

File: taskcluster/android_taskgraph/parameters.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+from taskgraph.parameters import extend_parameters_schema
+from voluptuous import All, Any, Range, Required
+
+
+def get_defaults(repo_root):
+    return {
+        "pull_request_number": None,
+        "release_type": "",
+    }
```
Comment (JohanLorenzo) [inline taskcluster/android_taskgraph/parameters.py:13]: Same rationale here, we may want to remove this function: 

https://github.com/mozilla-mobile/firefox-android/blob/191d5fd17d0a9d610770db7668ea6563d715effa/taskcluster/android_taskgraph/__init__.py#L38-L43

**Review by JohanLorenzo — APPROVED** [review]
LGTM modulo nits!

File: taskcluster/android_taskgraph/parameters.py (hunk 2 above)
Comment (ahal) [reply taskcluster/android_taskgraph/parameters.py:22]: Ouch, I didn't notice that even though I even touched that file!

Hmm, I wonder why running `taskgraph` locally didn't pick up those defaults? I agree that the way I have it implemented is the "official way" of doing it though.. So we can go with that.

**Comment by ahal** [comment]
Ah, it's failing because the `defaults_fn` method only applies when `strict=False` (e.g when running locally). So in CI it's still necessary to set the default in the `get_decision_parameters` function. There's definitely room for some cleanup in Taskgraph here.

File: taskcluster/android_taskgraph/parameters.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    defaults_fn=get_defaults,
+)
+
+
+def get_decision_parameters(graph_config, parameters):
+    # Environment is defined in .taskcluster.yml
+    pr_number = os.environ.get("MOBILE_PULL_REQUEST_NUMBER", None)
+    parameters["pull_request_number"] = None if pr_number is None else int(pr_number)
+    parameters.setdefault("next_version", None)
+    parameters.setdefault("release_type", "")
```
Comment (JohanLorenzo) [inline taskcluster/android_taskgraph/parameters.py:34]: Nit: Maybe not worth doing if we handle https://github.com/taskcluster/taskgraph/issues/235 soon enough. 

I'd recommend centralizing the default values in get_default. So `get_decision_parameters()` could look like this: 

```suggestion
    parameters |= get_defaults(graph_config.vcs_root))
    # Environment is defined in .taskcluster.yml
    pr_number = os.environ.get("MOBILE_PULL_REQUEST_NUMBER", None)
    parameters["pull_request_number"] = None if pr_number is None else int(pr_number)
```

File: taskcluster/test/params/main-repo-ship-release.yml [hunk 5]
```diff
@@ -149,15 +149,15 @@ existing_tasks:
...
 head_rev: 382516387821e0dc28bdc4699efe46554ebaef8b
-head_tag: v107.0.4
+head_tag: v112.2.0
 level: '3'
 moz_build_date: '20221031164327'
-next_version: 107.0.5
+next_version: 112.2.1
 optimize_target_tasks: true
-owner: jlorenzo@mozilla.com
+owner: user@example.com
```
Comment (JohanLorenzo) [inline taskcluster/test/params/main-repo-ship-release.yml:160]: 👍

**Review by JohanLorenzo — APPROVED** [review]
LGTM!

**Comment by ahal** [comment]
The task is failing due to the base image having Python 3.6 and Taskgraph requiring 3.7+. I may end up spinning this out into a separate PR.

**Review by gabrielBusta — APPROVED** [review]

**Pull request merged**